### DISCUSSION TITLE
How to define face dependent area emitter properties in mesh?

### DISCUSSION THREAD

**Discussion by hypevisenj** [mitsuba-renderer/mitsuba3#242] [Q&A]
Hello,

I want to model the appearance of an object that has blackbody emission with a non uniform temperature distribution where each mesh face has a temperature value

One possibility is to define an area emitter for each face in the mesh, but is there a better way to do this that is more concise?

**Comment by Speierers** [comment]
Hi @hypevisenj ,

You should be able to combine [`mesh-attribute`](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_textures.html#mesh-attribute-texture-mesh-attribute) textures and area light like this:

```xml
<shape type="ply">
  <string name="filename" value="my_mesh_with_face_color_attr.ply"/>
    <emitter type="area">
      <texture type="mesh_attribute" name="radiance">
        <string name="name" value="face_color"/>
      </texture>
   </emitter>
</shape>
```

Please let me know if that works for you.
